### ISSUE TITLE
Photino.Native.so: undefined symbol

### ISSUE THREAD

**Issue by KajiMaCN** [tryphotino/photino.NET#193]
Hi, I get the following error when I run the program, can you help me with this?

```
(base) mzh@openkylin:~/sn$ ./SmartNote.App
Photino.NET: "Photino".SetTitle(Photino.Blazor App)
Photino.NET: "Photino.Blazor App".SetUseOsDefaultSize(False)
Photino.NET: "Photino.Blazor App".SetUseOsDefaultLocation(False)
Photino.NET: "Photino.Blazor App".SetWidth(1000)
Photino.NET: "Photino.Blazor App".SetHeight(900)
Photino.NET: "Photino.Blazor App".SetLeft(0)
Photino.NET: "Photino.Blazor App".SetTop(100)
Photino.NET: "Photino.Blazor App".SetTitle(Smart Note)
Photino.NET: "Smart Note".SetMaximized(True)
Photino.NET: "Smart Note".SetDevTools(True)
Photino.NET: "Smart Note".SetContextMenuEnabled(True)
Photino.NET: "Smart Note".SetIconFile(./wwwroot/favicon.ico)
Photino.NET: "Smart Note".SetGrantBrowserPermission(True)
Photino.NET: "Smart Note".Load(/)
Photino.NET: "Smart Note" ** File "/" could not be found.
Photino.NET: "Smart Note".Load(app://localhost/)

(SmartNote.App:5078): GLib-GObject-CRITICAL **: 21:01:41.718: g_object_new_is_valid_property: object class 'WebKitSettings' has no property named 'enable_webrtc'
./SmartNote.App: symbol lookup error: /home/mzh/sn/Photino.Native.so: undefined symbol: webkit_website_data_manager_set_tls_errors_policy
```

**Comment by CypherPotato** [comment]
Tell us your operating system and environment details so we can help you better. I think that Photino relies on top of the operating system's native webview API, which may be older than the other and differ from OS to OS.

**Comment by MikeYeager** [comment]
@KajiMaCN It looks like you're on a Linux distro. Details would be very helpful. We use GTK and WebKitGTK, so it would be helpful to know the details of your distro. Alos, please see the Publish Photino sample in the samples repo. https://github.com/tryphotino/photino.Samples/tree/master/Photino.PublishPhotino/PublishPhotino

**Comment by KajiMaCN** [comment]
> @KajiMaCN It looks like you're on a Linux distro. Details would be very helpful. We use GTK and WebKitGTK, so it would be helpful to know the details of your distro. Alos, please see the Publish Photino sample in the samples repo. https://github.com/tryphotino/photino.Samples/tree/master/Photino.PublishPhotino/PublishPhotino

My system info is as follows:
Linux version 6.6.0-1-generic (buildd@1dbb9d95335a) (x86_64-linux-gnu-gcc (Openkylin 12.3.0-1ok3) 12.3.0, GNU ld (GNU Binutils for Openkylin) 2.41) #0ok8-KYLINOS SMP PREEMPT_DYNAMIC Thu Jun 13 06:55:27 UTC 2024

This OS is based on ubuntu development. We asked the developers and their system version dependencies are slow to update and there are many dependency conflicts. Currently we have replaced the solution and the program is working.Openkylin is a rather difficult operating system to use, so I don't recommend putting extra effort in supporting this system.

**Comment by MikeYeager** [comment]
@KajiMaCN Yes, that's the beauty and the curse of Linux. Lots of distros! If you take a look at the sample link above it's got a lot of great documentation about how to create deployments on Linux. You can extend it a bit and use the same techniques for setting up a development machine with the right installations to build and run. For dev machines, you can also take a look at the make file in the Photino.Native repo. It lists all of the libraries that need to be on the dev machine. I'm going to close this for now, but please re-open if we can be of further help.
